### PR TITLE
feat: add settings menu

### PR DESCRIPTION
**Pull request by pedromotita**
## Description

- add Settings Menu UI
- show Settings Menu via `.sheet` modifier

## Related Issues

- Closes #689

## How to test it

- Start any form of video
- Tap on the player to show the control buttons
- Tap on the gear icon to show the Settings Menu

### CODE REVIEW

**Comment by VitorPietrobom** [comment]
I feel like the settings button is a little too big, what do you think about making it smaller?

File: EnhancedMediaPlayer/SettingsMenu.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public var body: some View {
+        VStack(alignment: .leading, spacing: Constants.stackSpacing) {
+            renderSettingsItem()
+            renderSettingsItem()
+            renderSettingsItem()
+        }
+        .padding(Constants.stackSpacing)
+        .background(.background)
+        .cornerRadius(Constants.cornerRadius)
+    }
```
Comment (VitorPietrobom) [inline EnhancedMediaPlayer/SettingsMenu.swift:16]: I feel like this menu items should stick to the top of the thing that comes up...

**Review by VitorPietrobom — COMMENTED** [review]
Code wise this is looking good! Just left with the design thinks I commented

**Review by VitorPietrobom — APPROVED** [review]
LGTM!

File: EnhancedMediaPlayer/MediaPlayer.swift [hunk 2]
```diff
@@ -18,6 +18,16 @@ public struct MediaPlayer: UIViewControllerRepresentable {
     }
 
     public func updateUIViewController(_ mediaPlayerController: MediaPlayerViewController, context: Context) {
```
Comment (JPedroAmorim) [inline EnhancedMediaPlayer/MediaPlayer.swift:20]: Can't you annotate this with `@MainActor` instead?

File: EnhancedMediaPlayer/MediaPlayerView.swift [hunk 3]
```diff
@@ -16,16 +16,14 @@ public struct MediaPlayerView: View {
         ZStack {
             AVPlayerView(player: viewModel.player, playerState: viewModel.playerState)
             
-            if showActionButtons {
+            if viewModel.isShowingActionButtons {
```
Comment (JPedroAmorim) [inline EnhancedMediaPlayer/MediaPlayerView.swift:19]: ditto

File: EnhancedMediaPlayer/MediaPlayerView.swift [hunk 4]
```diff
@@ -16,16 +16,14 @@ public struct MediaPlayerView: View {
...
                 renderOverlay()
                 renderActionButtons()
                     .padding(.horizontal, screenWidth / Constants.paddingWidthDivider)
             }
         }
         .onTapGesture {
-            withAnimation {
```
Comment (JPedroAmorim) [inline EnhancedMediaPlayer/MediaPlayerView.swift:26]: ditto

File: EnhancedMediaPlayer/MediaPlayerView.swift [hunk 5]
```diff
@@ -86,9 +84,17 @@ public struct MediaPlayerView: View {
         }
     }
     
-    @ViewBuilder private func renderButton(control: Control, color: Color = .white) -> some View {
+    @ViewBuilder private func renderSettings() -> some View {
```
Comment (JPedroAmorim) [inline EnhancedMediaPlayer/MediaPlayerView.swift:87]: if the body of a function is literally one line, then you introduce more harm than good in creating it

File: EnhancedMediaPlayer/MediaPlayer.swift (hunk 2 above)
Comment (pedromotita) [reply EnhancedMediaPlayer/MediaPlayer.swift:17]: I made this particular implementation with @AmadeusK525 . Do you remember if there was any special reason to create the `extension`, or if it was just to make code smaller in `updateUIViewController`?

Comment (pedromotita) [reply EnhancedMediaPlayer/MediaPlayer.swift:17]: I tired marking the `updateUIViewConroller` method as `@MainActor` but I kept getting the warning. I removed the `MainActor` extension to make the code familiar to readers

File: EnhancedMediaPlayer/SettingsMenu.swift [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            renderSettingsItem()
+            renderSettingsItem()
+            renderSettingsItem()
+        }
+        .frame(maxHeight: .infinity, alignment: .topLeading)
+        .padding(Constants.stackPadding)
+    }
+
+    @ViewBuilder private func renderSettingsItem() -> some View {
+        HStack {
```
Comment (JPedroAmorim) [inline EnhancedMediaPlayer/SettingsMenu.swift:18]: implicit spacing

**Review by JPedroAmorim — APPROVED** [review]

**Pull request merged**